### PR TITLE
refac: Use client from wholesale NuGet

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## References

- https://app.zenhub.com/workspaces/mandalorian-5fd22a1a59e4740018f5ab5a/issues/energinet-datahub/opengeh-wholesale/17

### CODE REVIEW

File: apps/dh/api-dh/source/DataHub.WebApi/Registration/DomainRegistrationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+using Energinet.DataHub.MeteringPoints.Client.Extensions;
+using Energinet.DataHub.Wholesale.Client;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Energinet.DataHub.WebApi.Registration
+{
+    public static class DomainRegistrationExtensions
+    {
+        public static void AddDomainClients(this IServiceCollection services, ApiClientSettings apiClientSettings)
```
Comment (dstenroejl) [inline apps/dh/api-dh/source/DataHub.WebApi/Registration/DomainRegistrationExtensions.cs:28]: suggestion: "service collection extensions" should return the IServiceCollection to allow developers to chain methods.

File: apps/dh/api-dh/source/DataHub.WebApi/DataHub.WebApi.csproj [hunk 2]
```diff
@@ -35,6 +35,7 @@ limitations under the License.
   </Target>
 
   <ItemGroup>
+    <PackageReference Include="Energinet.DataHub.Wholesale.Client" Version="1.0.0-alpha-35" />
```
Comment (dstenroejl) [inline apps/dh/api-dh/source/DataHub.WebApi/DataHub.WebApi.csproj:38]: Currently a prerelease of this package is used - it should use a release.

File: apps/dh/api-dh/source/DataHub.WebApi/Registration/HealthEndpointRegistrationExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace Energinet.DataHub.WebApi.Registration
+{
+    public static class HealthEndpointRegistrationExtensions
+    {
+        public static IServiceCollection SetupHealthEndpoints(this IServiceCollection services, ApiClientSettings apiClientSettingsService)
+        {
+            const string liveEndpointPath = "/monitor/live";
+            var chargesLiveHealthUrl = apiClientSettingsService.ChargesBaseUrl == string.Empty
+                ? throw new ArgumentException()
```
Comment (jonasdmoeller) [inline apps/dh/api-dh/source/DataHub.WebApi/Registration/HealthEndpointRegistrationExtensions.cs:27]: Should these argument exceptions have a message that says which parameter is empty?

File: apps/dh/api-dh/source/DataHub.WebApi/Startup.cs [hunk 4]
```diff
@@ -110,10 +104,10 @@ public void ConfigureServices(IServiceCollection services)
                 });
             }
 
-            var apiClientSettings = Configuration.GetSection("ApiClientSettings").Get<ApiClientSettings>();
-            AddDomainClients(services, apiClientSettings);
+            var apiClientSettings = Configuration.GetSection("ApiClientSettings").Get<ApiClientSettings>()
+                                    ?? new ApiClientSettings();
```
Comment (dstenroejl) [inline apps/dh/api-dh/source/DataHub.WebApi/Startup.cs:108]: I'm not sure this is necessary. Reading the docs "Get<ApiClientSettings>" will return "default(T)" if settings cannot be binded.

**Review by dstenroejl — COMMENTED** [review]
Added a few suggestions. Notice we are referencing a prerelease of the wholesale client package, this should probably be a release version.

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/Controllers/ControllerTestsBase.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        protected HttpClient BffClient { get; }
+
+        protected ControllerTestsBase(
+            BffWebApiFixture bffWebApiFixture,
+            WebApiFactory factory,
+            ITestOutputHelper testOutputHelper)
+             : base(bffWebApiFixture, testOutputHelper)
+        {
+            DomainClientMock = new Mock<TDomainClient>();
+            BffClient = factory.WithWebHostBuilder(builder =>
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/Controllers/ControllerTestsBase.cs:43]: Thanks for the input. I haven't changed the functionality and will leave it to others to contribute to further improvements of the existing code.

File: apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/Controllers/ControllerTestsBase.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            {
+                builder.ConfigureServices(services =>
+                {
+                    services.AddTransient(_ => DomainClientMock.Object);
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi.Tests/Integration/Controllers/ControllerTestsBase.cs:47]: Thanks for the input. I haven't changed the functionality and will leave it to others to contribute to further improvements of the existing code.

File: apps/dh/api-dh/source/DataHub.WebApi/DataHub.WebApi.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi/DataHub.WebApi.csproj:38]: Thx. Fixed.

File: apps/dh/api-dh/source/DataHub.WebApi/Registration/DomainRegistrationExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        {
+            AddMeteringPointClient(services, apiClientSettings);
+            AddChargesClient(services, apiClientSettings);
+            AddMessageArchiveClient(services, apiClientSettings);
+            AddMarketParticipantClient(services, apiClientSettings);
+            AddWholesaleClient(services, apiClientSettings);
+
+            services.AddSingleton(apiClientSettings);
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi/Registration/DomainRegistrationExtensions.cs:36]: Thx. I've implemented your suggestion.

File: apps/dh/api-dh/source/DataHub.WebApi/Registration/DomainRegistrationExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        private static void AddWholesaleClient(IServiceCollection services, ApiClientSettings apiClientSettings)
+        {
+            var baseUri = GetBaseUri(apiClientSettings.WholesaleBaseUrl);
+            services.AddWholesaleClient(baseUri, AuthorizationHeaderProvider);
+        }
+
+        private static Uri GetBaseUri(string baseUri)
+        {
+            var emptyUrl = "https://empty";
+            var messageArchiveBaseUrl = Uri.TryCreate(baseUri, UriKind.Absolute, out var url)
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi/Registration/DomainRegistrationExtensions.cs:72]: Thx. Fixed.

File: apps/dh/api-dh/source/DataHub.WebApi/Registration/DomainRegistrationExtensions.cs (hunk 1 above)
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi/Registration/DomainRegistrationExtensions.cs:28]: Thx. Fixed.

File: apps/dh/api-dh/source/DataHub.WebApi/Startup.cs (hunk 4 above)
Comment (BjarkeMeier) [reply apps/dh/api-dh/source/DataHub.WebApi/Startup.cs:108]: But it is. Swagger file generation fails if I remove the null coalescing.

**Review by jonasdmoeller — APPROVED** [review]

**Review by dstenroejl — APPROVED** [review]

**Pull request merged**